hub/core/chunk_engine.py

```diff
@@ -0,0 +1,343 @@
+from hub.core.compression import decompress_array
+from math import ceil
+from typing import Optional, Sequence, Union, Tuple
+from hub.util.exceptions import (
+    CorruptedMetaError,
+    DynamicTensorNumpyError,
+    TensorDoesNotExistError,
+)
+from hub.core.meta.tensor_meta import TensorMeta
+from hub.core.index.index import Index
+from hub.util.keys import (
+    get_chunk_key,
+    get_chunk_id_encoder_key,
+    get_tensor_meta_key,
+)
+from hub.core.sample import Sample
+from hub.constants import DEFAULT_MAX_CHUNK_SIZE, UNCOMPRESSED
+
+import numpy as np
+
+from hub.core.storage.lru_cache import LRUCache
+
+from hub.core.chunk import Chunk
+
+from hub.core.meta.encode.chunk_id import ChunkIdEncoder
+
+
+SampleValue = Union[np.ndarray, int, float, bool, Sample]
+
+
+def is_uniform_sequence(samples):
+    if len(set(map(type, samples))) != 1:
```

Comment:
docstring please

Reply:
also this should either be a method or in case you foresee this being reused it should be in utils

Reply:
@AbhinavTuli neither of these are really true. It _could_ be a static method, but it's really just a one-off method used in `extend`. Where should functions like these live?